src/main/java/Entities/Player.java

```diff
@@ -109,6 +107,78 @@ public Player(String name, int money, boolean inJail, int jailCards, int positio
      */
     public boolean isInJail() { return inJail; }
 
+    /**
+     * Gets the number of railroads owned by this player
+     * @return the number of railroads owned by this player
+     */
+    public int getRailroads() {
+        int railroads = 0;
+        for (Property property : this.properties) {
+            if (Objects.equals(property.getColour(), "Railroad")) {
+                railroads += 1;
+            }
+        }
+        return railroads;
+    }
+
+    /**
+     * Gets the number of utilities owned by this player
+     * @return the number of utilities owned by this player
+     */
+    public int getUtilities() {
+        int utility = 0;
+        for (Property property : this.properties) {
+            if (Objects.equals(property.getColour(), "Utility")) {
+                utility += 1;
+            }
+        }
+        return utility;
+    }
+
+    /**
+     * Gets all the complete sets of properties owned by this player (sets are grouped by colours)
+     *
+     * @return an ArrayList of colours referring to the complete sets of properties owned by this player
+     */
+    public ArrayList<String> ownedPropertySets() {
+        ArrayList<String> ownedSets = new ArrayList<>();
+        HashMap<String, Integer> sets = createSetMap();
+        for (Property property : this.properties) {
+            sets.put(property.getColour(), sets.get(property.getColour()) + 1);
+            for (Map.Entry<String, Integer> colour : sets.entrySet()) {
```

Comment:
Using two for loops here may result in a set that has duplicated key values, but if it only works this way we can change this later.